source/docs/docs/identity-and-access-management/tenancy.md

```diff
@@ -0,0 +1,67 @@
+
+## Principles of Tenancy
+
+### Tenant Trees are Isolated
+
+Tenants provide isolated access to data. Access to data within a tenant tree needs to be explicitly granted.
+
+#### Rationale
+
+Tenant isolation limits access privileges to data and control over resources managed by the ecosystem.
+
+#### Implications
+
+Access to a parent tenant is not implied by granting access to a child tenant and access to one tenant does not imply access to other unrelated tenant trees. [Access granted to a parent tenant *does* imply granting access to child tenants](#tenant-permissions-are-inherited)
+
+### Tenant permissions are inherited
+
+Access granted on a tenant will be inherited by all subtenants.
+
+#### Rationale
+
+Inheriting access rights from the parent is more intuitive and fits better into existing authorization paradigms.
+
+#### Implication
+
+When access is granted to a tenant, that access will be granted to all subtenants of the tenant at the same level. Access cannot be limited to only one level unless the tenant has no children. In some cases, a flat tenant structure may be better serve isolation requirements.
+
+### Tenants are Hierarchical
+
+Tenants form a tree structure, and are not limited to a single level.
+
+#### Rationale
+
+Experience with multiple large projects (Kubernetes, Openstack, etc) has shown that many ecosystems start with single level tenancy and then migrate to hierarchical tenancy in the future. This migration is often fraught with peril, bugs and dragons.
+
+#### Implications
+
+Tenants can have sub-tenants that can have sub-tenants, etc. Recursion needs to be accounted for in services built for the Infratographer ecosystem.
+
+### Services in the ecosystem are Multi-Tenant
+
+Services and applications in the Infratographer ecosystem will need to interact with multiple tenants. This will be done securely and with awareness.
+
+#### Rationale
+
+To support multitenancy, ecosystem services need to be tenant aware or have an instance deployed for every tenant. The former is preferable to the later.
+
+#### Implications
+
+The services that form the Infratographer ecosystem are aware of the tenants and provide the means of segmenting resources owned by those tenants. This allows for a single application to be provided to multiple tenants. Resource segmentation must be explicit.
+
+### Resources are owned by Tenants
+
+All resources are owned by a single tenant, not multiple tenants.
+
+#### Rationale
+
+Granular access control will provide for a more secure environment. Identities can be given access to individual resources or tenants. Tracking an arbitrary number of tenants for a resource adds unecessary complexity.
+
+#### Implications
+
+Services in the Infratographer ecosystem must track a single tenant association for each resource they manage.
```

Comment:
Another implication I'd like to add is that from a model view. There should be no difference between a "tenant" and a "subtenant" other than the fact that one has no parent in the tree. Having a distinction in these entities leads to complications when needing to apply "global" (or at least for a whole tree) authorization constructs and quota management

Reply:
I'll add this as an implication for now, but I think this may be a principle on its own. What do you think? Can you help me phrase one if you agree?